src/appstore-reviews.ts

```diff
@@ -0,0 +1,235 @@
+        }
+
+        if (regions.length == 0) {
+            throw "At least one region should be define in configuration"
+        }
+
+        const requests = (regions as string[]).map(region => this.fetchAppStoreReviews(id, config.pageRange || 5, region))
+        const result = await Promise.all(requests)
+
+        let reviews: Review[] = []
+        var newReviews = reviews.concat(...result).filter((review) => {
+            if (publishedReviews[id] !== undefined && publishedReviews[id].indexOf(review.id) >= 0) {
+                return false
+            }
+            return true
+        })
+
+        var newReviewsMap: PublishedReviews = {}
+        newReviewsMap[id] = newReviews.map(this.mapReviewId)
+
+        return {
+            messages: newReviews.map(review => {
+                if (config.generateMessageFromReview !== undefined) {
+                    return config.generateMessageFromReview.call(this, review, appInformation, config)
+                } else {
+                    return this.generateSlackMessage(review, appInformation, config)
+                }
+            }),
+            newReviews: newReviewsMap
+        }
+    }
+
+    async fetchAppInformation(appId: string): Promise<AppInformation> {
+        const url = `${this.BASE_URL}/lookup?id=${appId}`
+        try {
+            const res = await axios.get(url) as AppInformationResponse
+            const results = res.data.results
+
+            if (results === null || results.length == 0) {
+                throw `Couldn't find app[${appId}] information`
+            }
+            const result = results[0]
+
+            return {
+                appName: result.trackCensoredName,
+                appIcon: result.artworkUrl100,
+                appLink: result.trackViewUrl
+            }
+
+        } catch (err) {
+            console.error(`Something went wrong, ${err}`)
+            throw err
+        }
+    }
+
+    async fetchAppStoreReviews(appId: string, pagesInRange: number, region: string) {
+        let reviews: Review[] = []
+
+        for (let page = 1; page <= pagesInRange; page++) {
+            let pageReview = await this.fetchAppStoreReviewsByPage(appId, page, region)
+            reviews = reviews.concat(pageReview)
+        }
+
+        return reviews
+    }
+
+    async fetchAppStoreReviewsByPage(appId: string, page: number, region: string): Promise<Review[]> {
+        const url = `${this.BASE_URL}/${region}/rss/customerreviews/page=${page}/id=${appId}/sortBy=mostRecent/json`
+        try {
+            const res = await axios.get(url) as ReviewResponse
+            var entries = res.data.feed.entry
+
+            if (entries == null || entries.length == 0) {
+                return []
+            }
+
+            return entries
+                .filter(this.isAppInformationEntry)
+                .reverse()
+                .map((review) => this.parseAppStoreReview(review, region))
+
+        } catch {
+            return []
```

Comment:
Logging the error that caused the exception would help debugging later on.

Reply:
Opps, I missed it